bundles/org.openhab.binding.mikrotik/src/main/resources/OH-INF/thing/thing-types.xml

```diff
@@ -0,0 +1,431 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<thing:thing-descriptions bindingId="mikrotik"
+	xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+	xmlns:thing="https://openhab.org/schemas/thing-description/v1.0.0"
+	xsi:schemaLocation="https://openhab.org/schemas/thing-description/v1.0.0 https://openhab.org/schemas/thing-description-1.0.0.xsd">
+
+	<bridge-type id="routeros">
+		<label>Mikrotik RouterOS</label>
+		<description>A connection to RouterOS device</description>
+		<channels>
+			<channel id="freeSpace" typeId="freeSpace"/>
+			<channel id="totalSpace" typeId="totalSpace"/>
+			<channel id="usedSpace" typeId="usedSpace"/>
+			<channel id="freeMemory" typeId="freeMemory"/>
+			<channel id="totalMemory" typeId="totalMemory"/>
+			<channel id="usedMemory" typeId="usedMemory"/>
+			<channel id="cpuLoad" typeId="cpuLoad"/>
+			<channel id="uptime" typeId="uptime"/>
+			<channel id="upSince" typeId="upSince"/>
+		</channels>
+		<properties>
+			<property name="vendor">Mikrotik</property>
+			<property name="modelId">RouterOS</property>
+			<property name="firmware"/>
+			<property name="serial"/>
+		</properties>
+		<representation-property>name</representation-property>
+		<config-description>
+			<parameter name="host" type="text" required="true">
+				<label>Hostname</label>
+				<description>Hostname or IP address of the RouterOS device</description>
+				<default>192.168.88.1</default>
+				<context>network-address</context>
+			</parameter>
+			<parameter name="port" type="integer" max="65535" min="1" required="false">
+				<label>API Port</label>
+				<description>API Port number of the RouterOS device</description>
+				<default>8728</default>
+			</parameter>
+			<parameter name="login" type="text" required="true">
+				<label>Username</label>
+				<default>admin</default>
+				<description>The username to access the the RouterOS device</description>
+			</parameter>
+			<parameter name="password" type="text" required="true">
+				<label>Password</label>
+				<description>The user password to access the RouterOS device</description>
+				<context>password</context>
+			</parameter>
+			<parameter name="refresh" type="integer" required="false">
```

Comment:
```suggestion
			<parameter name="refresh" type="integer" required="true">
```
Is it valid to leave it blank?

Reply:
Absolutely

Reply:
What happens in your binding if someone deletes the default 10 from the field and leaves it blank? By making it required="true" it gets highlighted RED in colour and lets the user know the field is needed and wont accept a blank entry. Feel free to leave it as is if the binding handles it when someone deletes all values.

Reply:
I've just checked - when I delete any config value (required or optional) with the default declared and then move focus away from the field, the default value is set back by the frontend. It feels like currently there is no way to leave it blank.